pkg/policy/rule.go

```diff
@@ -217,7 +214,28 @@ func mergeL4Ingress(ctx *SearchContext, rule api.IngressRule, ruleLabels labels.
 		}
 	}
 
+	var (
+		cnt int
+		err error
+	)
+
+	// L3-only rule without requirements.
+	if len(rule.ToPorts) == 0 && len(rule.FromRequires) == 0 {
```

Comment:
nit on function naming: this now makes `mergeL4Ingress()` include L3 which is a bit misleading. Same for egress side.

Reply:
Fixed - updated a bunch of other functions which included 'L4' in them as well. I specified port / proto as the thing being 'merged' since L3-only equates to '0/ANY' as a 'port-protocol' tuple. LMK if you think another naming convention is better.